### PR TITLE
Much better API for auth.{cpp, h}

### PR DESCRIPTION
**Pull request by est31**
auth.cpp has a very hard to understand API. Its one of my earliest contributions to minetest, and while I am proud of the other SRP stuff, I'm not very proud of this particular API design.

This patch greatly improves the API.

### CODE REVIEW

File: src/util/auth.cpp [hunk 1]
```diff
@@ -46,7 +46,11 @@ std::string translatePassword(const std::string &name,
 	return pwd;
 }
 
-void getSRPVerifier(const std::string &name,
+// Call lower level SRP code to generate a verifier with the
+// given pointers. Contains the preparations, call parameters
+// and error checking common to all srp verifier generation code.
+// See docs of srp_create_salted_verification_key for more info.
```
Comment (ShadowNinja) [inline src/util/auth.cpp]: Perhaps make this a Doxygen comment?

**Comment by est31** [comment]
@ShadowNinja is that your approval, or do you only track mine as creator of the PR?

File: src/util/auth.cpp (hunk 1 above)
Comment (est31) [inline src/util/auth.cpp]: Its a file local method, I don't think anybody will benefit from a doxygen comment.

**Comment by ShadowNinja** [comment]
I meant that I approve this.  :-)

**Comment by est31** [comment]
e0151d1054c9923630b8ae9cdae36efa261756d3

**Pull request closed without merging**